src/database/postgres_test.go

```diff
@@ -0,0 +1,253 @@
+package database
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"reflect"
+	"strings"
+	"testing"
+	"time"
+
+	pb "accretional.com/semantifly/proto/accretional.com/semantifly/proto"
+	"github.com/go-pg/pg/v10"
+	"github.com/jackc/pgx/v5"
+	_ "github.com/lib/pq"
+	"github.com/stretchr/testify/assert"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+func setupPostgres() error {
+	err := os.Chdir("../..")
+	if err != nil {
+		return fmt.Errorf("Failed to change directory: %v", err)
+	}
+
+	cmd := exec.Command("bash", "setup_postgres.sh")
+
+	var stdout, stderr bytes.Buffer
+	cmd.Stdout = &stdout
+	cmd.Stderr = &stderr
+
+	err = cmd.Run()
+	if err != nil {
+		return fmt.Errorf("Failed to setup PostgreSQL server: %v", err)
+	}
+
+	return nil
+}
+
+func createTestingDatabase() (*pg.DB, error) {
+	// Connect to the default "postgres" database
+	db := pg.Connect(&pg.Options{
+		User:     "postgres",
+		Password: "postgres",
+		Addr:     "localhost:5432",
+		Database: "postgres",
+	})
+
+	// Drop the database if it exists, then create it
+	_, err := db.Exec("DROP DATABASE IF EXISTS testdb")
+	if err != nil {
+		db.Close()
+		return nil, fmt.Errorf("failed to drop existing test database: %v", err)
+	}
+
+	_, err = db.Exec("CREATE DATABASE testdb")
+	if err != nil {
+		db.Close()
+		return nil, fmt.Errorf("failed to create test database: %v", err)
+	}
+
+	// Close the connection to the "postgres" database
+	db.Close()
+
+	// Connect to the newly created database
+	testDB := pg.Connect(&pg.Options{
+		User:     "postgres",
+		Password: "postgres",
+		Addr:     "localhost:5432",
+		Database: "testdb",
+	})
+
+	return testDB, nil
+}
+
+func removeTestingDatabase() error {
+	// Connect to the default "postgres" database to drop the test database
+	defaultDB := pg.Connect(&pg.Options{
+		User:     "postgres",
+		Password: "postgres",
+		Addr:     "localhost:5432",
+		Database: "postgres",
+	})
+	defer defaultDB.Close()
+
+	// Terminate all connections to the test database
+	_, err := defaultDB.Exec("SELECT pg_terminate_backend(pid) FROM pg_stat_activity WHERE datname = 'testdb'")
+	if err != nil {
+		return fmt.Errorf("Failed to terminate connections to test database: %v", err)
+	}
+
+	// Drop the test database
+	_, err = defaultDB.Exec("DROP DATABASE IF EXISTS testdb")
+	if err != nil {
+		return fmt.Errorf("Failed to drop test database: %v", err)
+	}
+
+	return nil
+}
+
+func TestPostgres(t *testing.T) {
+	const maxRetries = 3
+	var err error
+
+	if err := setupPostgres(); err != nil {
+		t.Fatalf("Failed to setup Postgres server: %v", err)
+	}
+
+	// Set a mock DATABASE_URL for testing
+	os.Setenv("DATABASE_URL", "postgres://postgres:postgres@localhost:5432/testdb")
+	defer os.Unsetenv("DATABASE_URL")
+
+	var db *pg.DB
+	for i := 0; i < maxRetries; i++ {
```

Comment:
This seems like a hacky workaround of a bug, race condition, or flake. Why did you add this and is it truly necessary? You can probbaly configure the client connection to wait for the db to be ready to handle failures caused by attempting to connect to the db too early during its startup

Reply:
Yeah this is for the race condition for the test to wait until the db is set up. I do need to wait a few seconds for the db to be ready.

Reply:
Removed it